src/rdf4cpp/rdf/datatypes/registry/DatatypeRegistry.hpp

```diff
@@ -322,6 +323,150 @@ class DatatypeRegistry {
      * or there is no viable conversion, else the found conversion
      */
     [[nodiscard]] static std::optional<DatatypeConverter> get_common_type_conversion(DatatypeIDView lhs_type_id, DatatypeIDView rhs_type_id) noexcept;
+
+public:
+    class LangTagInlines {
+        /**
+         * language tags to try to inline.
+         * the vector can be modified at startup.
+         * WARNING: modifying the vector after any Literal was created is UNDEFINED BEHAVIOR.
+         * this includes Literals already created in a persistent node storage.
+         */
+#ifdef MODIFYABLE_LANG_TAG_INLINES
+        static inline std::vector<std::string>
+                tags_to_inline{
+                        "de",
+                        "en",
+                        "fr",
+                        "ch",
+                };
+
+    public:
+        /**
+         * sets the language tags to try to inline.
+         * may only be called at startup.
+         * WARNING: calling this method after any Literal was created is UNDEFINED BEHAVIOR.
+         * this includes Literals already created in a persistent node storage.
+         */
+        static inline void set_tags_to_inline(std::initializer_list<std::string> li) {
+            tags_to_inline = li;
+        }
+        /**
+         * add one language tag to try to inline.
+         * may only be called at startup.
+         * WARNING: calling this method after any Literal was created is UNDEFINED BEHAVIOR.
+         * this includes Literals already created in a persistent node storage.
+         */
+        static inline void add_tag_to_inline(std::string_view t) {
+            tags_to_inline.emplace_back(t);
+        }
+#else
+        static constexpr std::array<std::string_view, 4> tags_to_inline{
+                "de",
+                "en",
+                "fr",
+                "ch",
+        };
+
+    public:
+        /**
+         * define MODIFYABLE_LANG_TAG_INLINES to modify which tags get inlined at runtime.
+         */
+        static void set_tags_to_inline(std::initializer_list<std::string>) = delete;
+        /**
+         * define MODIFYABLE_LANG_TAG_INLINES to modify which tags get inlined at runtime.
+         */
+        static void add_tag_to_inline(std::string_view) = delete;
+#endif
+
+    public:
+        static inline const auto &get_tags_to_inline() {
+            return tags_to_inline;
+        }
+
+        using LangTagID = uint64_t;
+
+        static constexpr size_t bits_needed_for(uint64_t i) {
+            return std::bit_width(i);
+        }
+
+        /**
+         * number of bits needed for the current tags to inline
+         */
+        static inline size_t inlined_size() noexcept {
```

Comment:
There are some functions here that I don't think should be publicly accessible, as they are kind of implementation details. This is one of them, but there are a few more below here. (everything related to the bit fiddling).

Also: it probably makes sense to mark all these `constexpr`, compiler probably treats them as such anyways but doesn't hurt. (also no need for `inline` then)

Reply:
there is currently a test case for `inlined_size()`, everything else is private now